examples/DivisorCurve.jl

```diff
@@ -0,0 +1,329 @@
+################################################################################
+# Sum of two divisors
+
+function Base.:+(D::AffineCurveDivisor, E::AffineCurveDivisor)
+    _check_same_curve(D, E) || error("The divisors are not on the same curve")
+    return AffineCurveDivisor(D.C, mergewith(+, D.divisor, E.divisor))
+end
+
+function Base.:+(D::ProjCurveDivisor, E::ProjCurveDivisor)
+    _check_same_curve(D, E) || error("The divisors are not on the same curve")
+    return ProjCurveDivisor(D.C, mergewith(+, D.divisor, E.divisor))
+end
+
+################################################################################
+
+function Base.:-(D::CurveDivisor, E::CurveDivisor)
+    return D + -1*E
+end
+
+################################################################################
+################################################################################
+
+function Base.:*(k::Int, D::AffineCurveDivisor{S}) where S <: FieldElem
+    if k == 0
+        return AffineCurveDivisor(D.C, Dict{Point{S}, Int}())
+    elseif k == 1
+        return D
+    else
+        return AffineCurveDivisor(D.C, Dict((P, k*m) for (P, m) in D.divisor))
+    end
+end
+
+function Base.:*(k::Int, D::ProjCurveDivisor{S}) where S <: FieldElem
+    if k == 0
+        return ProjCurveDivisor(D.C, Dict{Oscar.Geometry.ProjSpcElem{S}, Int}())
+    elseif k == 1
+        return D
+    else
+        return ProjCurveDivisor(D.C, Dict((P, k*m) for (P, m) in D.divisor))
+    end
+end
+
+################################################################################
+
+function ==(D::CurveDivisor, E::CurveDivisor)
+    return D.C == E.C && D.divisor == E.divisor
+end
+
+################################################################################
+################################################################################
+
+@doc Markdown.doc"""
+    iseffective(D::CurveDivisor)
+
+Return `true` if `D` is an effective divisor, `false` otherwise.
+"""
+function iseffective(D::CurveDivisor)
+    return all(v >= 0 for v in values(D.divisor))
+end
+
+################################################################################
+################################################################################
+
+@doc Markdown.doc"""
+    multiplicity(C::AffinePlaneCurve, phi::AbstractAlgebra.Generic.Frac, P::Point)
+
+Return the multiplicity of the rational function `phi` on the curve `C` at the point `P`.
+"""
+function multiplicity(C::AffinePlaneCurve, phi::AbstractAlgebra.Generic.Frac, P::Point)
+    f = divrem(phi.num, C.eq)
+    g = divrem(phi.den, C.eq)
+    !iszero(g[2]) || error("This is not a rational function on `C`")
+    if iszero(f[2])
+        return Inf
+    elseif isconstant(f[2])
+        a = 0
+    else
+        a = intersection_multiplicity(C, AffinePlaneCurve(f[2]), P)
+    end
+    if isconstant(g[2])
+        b = 0
+    else
+        b = intersection_multiplicity(C, AffinePlaneCurve(g[2]), P)
+    end
+    return a-b
```

Comment:
Perhaps I am missing something, but: can't this be simplified to something like this, relying on the next method?
```suggestion
    return multiplicity(C, f, P) -  multiplicity(C, g, P)
```

Reply:
Indeed, thanks !